STYLE.md

```diff
@@ -0,0 +1,228 @@
+# Introduction
+All code needs to be formatted according to .pre-commit-config.yaml. Each pull request is checked by [Formatting Tests](https://github.com/Azure/azure-osconfig/blob/dev/.github/workflows/formatting.yml).
+Apart from that there are style rules that can't be expressed in clang-format but must be followed nevertheless:
+
+# Formatting quirks
+## Boolean statements (e.g. in if) should explicitly use parenthesis and never rely on operators order:
+  - Bad: `(a != b && c != d)`
+  - Good: `((a != b) && (c != d))`
+## In comparison statements the constant should be on the left of the comparison operator
+  - Bad: `if (variable == 0)`
+  - Good: `if (0 < variable)`
+# Readability
+- Always chose readability over compactness
+- Avoid side-effects in if statements, never use side-effects in if statements if they are conditional:
+    - Bad:
+
+            if (NULL == (x = malloc(10))) // Avoid
+            {
+            }
+
+            if ((7 == a) && (NULL == (y = malloc(10)))) // Must not use
+            {
+            }
+
+
+    - Good:
+
+            x = malloc(10);
+            if (NULL == x)
+            {
+            }
+
+            if (7 == a)
+            {
+              y = malloc(10);
+              if (NULL == y)
+              {
+              }
+            }
+
+- Avoid ternary operators with complex statements
+- Avoid deeply nested if statements, use helper functions, early returns or goto to cleanup code:
```

Comment:
>Avoid deeply nested if statements, use helper functions, early returns or goto to cleanup code: [](http://example.com/codeflow?start=2&length=95)

This is not inline with existing code, with 2 exceptions:

1. The NRP code which follows a slightly different coding style due to how it was generated and modeled the first time. Also because of the code there the pattern to 'if (falure) then goto cleanup' works best.

2. The existing code does early returns for input parameter validation. We need to anchor this one in the STYLE doc.